parser/coldef/col_def.go

```diff
@@ -103,8 +103,11 @@ func getDefaultValue(c *ConstraintOpt, tp byte, fsp int) (interface{}, error) {
 
 		return value, nil
 	}
-
-	return expression.FastEval(c.Evalue), nil
+	v := expression.FastEval(c.Evalue)
+	if vv, ok := v.(*types.DataItem); ok {
```

Comment:
maybe abstract a function naming IsDateItem may be better for other places check.